File: MyProjects/MAP/Assignment 6/src/Model/Statements/CompStmt.java
package Model.Statements;

import Model.ProgramState;
import Model.Utilities.IStack;

import java.io.IOException;

public class CompStmt implements IStmt {
    private  IStmt first, second;

    public CompStmt(IStmt f, IStmt s){
        first = f;
        second = s;
    }

    @Override
    public ProgramState execute(ProgramState state){
        IStack<IStmt> stack = state.getStack();
        stack.add(second);
        stack.add(first);

        return null;
    }

    public String toString(){
        return "(" + first + "; " + second + ")";
    }
}

File: MyProjects/MAP/Assignment 7/Interpreter/Model/Statements/CompStmt.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Interpreter.Model.Statements
{
    public class CompStmt:IStmt
    {
        private IStmt first, second;

        public CompStmt(IStmt f, IStmt s)
        {
            first = f;
            second = s;
        }

        public ProgramState execute(ProgramState state)
        {
            state.Stack.add(second);
            state.Stack.add(first);

            return state;
        }

        public override string ToString()
        {
            return "(" + first + "; " + second + ")";
        }
    }
}
